api/include/opentelemetry/context/context.h

```diff
@@ -133,8 +130,17 @@ class Context
       key_        = new char[key.size()];
       key_length_ = key.size();
       memcpy(key_, key.data(), key.size() * sizeof(char));
-      value_ = value;
       next_  = nostd::shared_ptr<DataList>{nullptr};
+      value_ = value;
+    }
+
+    DataList(const DataList &other)
```

Comment:
Ok to have a copy constructor, this clears the cppcheck warning, and is safer in case of refactoring.

I suspect this code to be unreachable.